src/FSharpVSPowerTools.Core/UntypedAstUtils.fs

```diff
@@ -9,7 +9,7 @@ type internal ShortIdent = string
 type internal Idents = ShortIdent[]
 
 let internal longIdentToArray (longIdent: LongIdent): Idents =
-    longIdent |> List.map string |> List.toArray
+    Array.init longIdent.Length (fun i -> string longIdent.[i])
```

Comment:
I agree with @dungpa, except that `longIdent.Length` is called only once. The real problem here is `longIdent.[i]` which is O(n).